### DISCUSSION TITLE
how is TTS usage determined?

### DISCUSSION THREAD

**Discussion by DarthZyll** [rkusa/DATIS#138] [Q&A] [answered]
Hi team,

Great little product, our team uses it over on TAW. We just recently activated our free Google Cloud TTS API, and I had a question about how DATIS determines overall usage. If we have 3 DATIS static objects placed, and they each speak 100 words at ~5 bytes each (just as an example), does this mean that each time the mission starts up, it will use 3x100x5=1500bytes only, or does it use the API each and every time someone tunes into the freq, and even when the message starts looping itself? I assume the former, but wanted to be sure.

And if the mission were cycling in a rotation (let's say the mission restarts every hour), is it also safe to assume that a fresh API hit occurs when the mission restarts, so in this example, 1500bytes every hour?

Thanks in advance,
Zyll - TAW

**Comment by rkusa** [comment, marked as answer]
Hi,

> each time the mission starts up, it will use 3x100x5=1500bytes only

yes, correct. The API is only called when the report content changed (including the first report when the mission starts). The only reason a report might change is right now:
- new ATIS information letter after 60min,
- weather changes due to dynamic weather or a unit that moved to a location with slightly different weather (or different BRC for carriers).

> And if the mission were cycling in a rotation (let's say the mission restarts every hour), is it also safe to assume that a fresh API hit occurs when the mission restarts, so in this example, 1500bytes every hour?

Correct

Cheers
